src/core/oned/rss/RSS14Reader.ts

```diff
@@ -121,67 +121,71 @@ export default class RSS14Reader extends AbstractRSSReader {
 
     private decodePair(row: BitArray, right: boolean, rowNumber: number, hints: Map<DecodeHintType, any>): Pair {
         try {
-            let startEnd = this.findFinderPattern(row, right);
-            let pattern = this.parseFoundFinderPattern(row, rowNumber, right, startEnd);
+            const startEnd = this.findFinderPattern(row, right);
+            const pattern = this.parseFoundFinderPattern(row, rowNumber, right, startEnd);
 
-            let resultPointCallback = hints == null ? null : <ResultPointCallback>hints.get(DecodeHintType.NEED_RESULT_POINT_CALLBACK);
+            const resultPointCallback = hints == null ? null : <ResultPointCallback>hints.get(DecodeHintType.NEED_RESULT_POINT_CALLBACK);
 
             if (resultPointCallback != null) {
-                let center = (startEnd[0] + startEnd[1]) / 2.0;
+                let center = (startEnd[0] + startEnd[1] - 1) / 2.0;
                 if (right) {
                     // row is actually reversed
                     center = row.getSize() - 1 - center;
                 }
                 resultPointCallback.foundPossibleResultPoint(new ResultPoint(center, rowNumber));
             }
 
-            let outside = this.decodeDataCharacter(row, pattern, true);
-            let inside = this.decodeDataCharacter(row, pattern, false);
+            const outside = this.decodeDataCharacter(row, pattern, true);
+            const inside = this.decodeDataCharacter(row, pattern, false);
             return new Pair(1597 * outside.getValue() + inside.getValue(),
                 outside.getChecksumPortion() + 4 * inside.getChecksumPortion(),
                 pattern);
         }
-        catch (err) {
-            return null;
+        catch (ex) {
+            if (ex instanceof NotFoundException) {
+                return null;
+            } else {
+                throw ex;
+            }
         }
     }
 
     private decodeDataCharacter(row: BitArray, pattern: FinderPattern, outsideChar: boolean): DataCharacter {
 
-        let counters = this.getDataCharacterCounters();
+        const counters = this.getDataCharacterCounters();
         for (let x = 0; x < counters.length; x++) {
             counters[x] = 0;
         }
 
         if (outsideChar) {
             OneDReader.recordPatternInReverse(row, pattern.getStartEnd()[0], counters);
         } else {
-            OneDReader.recordPattern(row, pattern.getStartEnd()[1] + 1, counters);
+            OneDReader.recordPattern(row, pattern.getStartEnd()[1], counters);
```

Comment:
how so?

Reply:
This is taken directly from https://github.com/zxing/zxing/pull/1179. If I understand correctly, they argue the current implementation returns a result which claims the pattern is a single pixel off where the pattern actually starts/ends.